src/tdastro/effects/redshift.py

```diff
@@ -0,0 +1,76 @@
+from tdastro.base_models import EffectModel
+
+
+class Redshift(EffectModel):
+    """A redshift effect model.
+
+    This contains a "pre-effect" method, which is used to calculate the emitted wavelengths/times
+    needed to give us the observed wavelengths and times given the redshift.
+
+    Attributes
+    ----------
+    pz : `float`
+        The redshift.
+
+    Notes
+    -----
+    Conversions used are as follows:
+    - emitted_wavelength = observed_wavelength / (1 + redshift)
+    - emitted_time = observed_time / (1 + redshift)
+    - observed_flux = emitted_flux / (1 + redshift)
+    """
+
+    def __init__(self, pz, **kwargs):
+        """Create a Redshift effect model.
+
+        Parameters
+        ----------
+        pz : `float`
+            The redshift.
+        **kwargs : `dict`, optional
+           Any additional keyword arguments.
+        """
+        super().__init__(**kwargs)
+        self.pz = pz
+
+    def pre_effect(self, observed_times, observed_wavelengths, **kwargs):
+        """Calculate the emitted wavelengths/times needed to give us the observed wavelengths
+        and times given the redshift.
+
+        Parameters
+        ----------
+        observed_times : float
+            The times at which the observation is made.
+        observed_wavelengths : float
+            The wavelengths at which the observation is made.
+        **kwargs : `dict`, optional
+           Any additional keyword arguments.
+
+        Returns
+        -------
+        float
+            The adjusted flux density at the observed wavelength.
+        """
+        return (observed_times / (1 + self.pz), observed_wavelengths / (1 + self.pz))
```

Comment:
I think we want to make it a bit more complicated with time. Our "time" is actually a "date", so we cannot multiply it - the result would have little sense. Also, it is not convenient to define time in the model to be "deredshifted" date, because than we cannot generate reference epoch (for example peak date or event start date) without knowing a redshift.

I think the only solution is to ask the model to define the "epoch" (reference time) which would be the time (date) in observer frame, and use it here. So all the times in the rest frame would be relative to this reference time:
```python
rest_times = (observed_times - epoch) / (1 + self.pz) + epoch
```

Reply:
Notes from convo:

It should be fine to assume every PhysicalModel to have an epoch. (This is straightforward for, say, a supernova with a peak; for a variable star, Kostya says one would use one of the peaks. For say, an AGN, one might just define the epoch as the beginning of the survey)